spec/models/product_spec.rb

```diff
@@ -0,0 +1,292 @@
+RSpec.describe Product, type: :model do
+    brand = Brand.first
+    
+
+
+    describe '#title' do
+        it 'has no title' do
+            product = Product.new(
+                brand: brand,
+                title: nil,
+                price: 20000,
+                quantity: 1,
+                description: "a" * 20 
+            )
+            expect(product).to_not be_valid
+        end
+
+        it 'title minimum length' do
+            product = Product.new(
+                brand: brand,
+                title: "a" * 5,
+                price: 20000,
+                quantity: 1,
+                description: "a" * 20 
+            )
+            expect(product).to be_valid
+        end
+
+        it 'title maximum length' do
+            product = Product.new(
+                brand: brand,
+                title: "a" * 30,
+                price: 20000,
+                quantity: 1,
+                description: "a" * 20 
+            )
+            expect(product).to be_valid
+        end
+
+        it 'title minimum length lower bound check' do
+            product = Product.new(
+                brand: brand,
+                title: "a" * 4,
+                price: 20000,
+                quantity: 1,
+                description: "a" * 20 
+            )
+            expect(product).to_not be_valid
+        end
+
+        it 'title minimum length upper bound check' do
+            product = Product.new(
+                brand: brand,
+                title: "a" * 6,
+                price: 20000,
+                quantity: 1,
+                description: "a" * 20 
+            )
+            expect(product).to be_valid
+        end
+
+        it 'title maximum length lower bound check' do
+            product = Product.new(
+                brand: brand,
+                title: "a" * 29,
+                price: 20000,
+                quantity: 1,
+                description: "a" * 20 
+            )
+            expect(product).to be_valid
+        end
+
+        it 'title maximum length upper bound check' do
+            product = Product.new(
+                brand: brand,
+                title: "a" * 31,
+                price: 20000,
+                quantity: 1,
+                description: "a" * 20 
+            )
+            expect(product).to_not be_valid
+        end
+
+        it 'title Uniqueness check' do
+            product1 = Product.new(
+                brand: brand,
+                title: "a" * 30,
+                price: 20000,
+                quantity: 1,
+                description: "a" * 20 
+            )
+            product1.save
+            product2 = Product.new(
+                brand: brand,
+                title: "A" * 30,
+                price: 20000,
+                quantity: 1,
+                description: "a" * 20 
+            )
+            expect(product2).to_not be_valid
+        end
+
+    end
+
+
+
+    describe '#Description' do
+        it 'has no Description' do
```

Comment:
Kindly use descriptive message that tells what is being tested. e.g. 'It gives error if no description is provided'

Reply:
Okay got it.